src/main/java/frc/robot/subsystems/climber/ClimberHardware.java

```diff
@@ -0,0 +1,122 @@
+
+        private final SparkMaxConfig leftClimberConfig = new SparkMaxConfig();
+        private final SparkMaxConfig rightClimberConfig = new SparkMaxConfig();
+
+        private final RelativeEncoder leftClimberEncoder = leftClimber.getEncoder();
+        private final SparkClosedLoopController climberClosedLoopController = leftClimber.getClosedLoopController();
+
+        private Angle climberGoalAngle = Radians.of(0.0);
+        private Angle servoGoalAngle = Radians.of(0.0);
+
+        public ClimberHardware() {
+
+                leftClimberConfig.inverted(ClimberConstants.LEFT_CLIMBER_INVERTED)
+                                .idleMode(ClimberConstants.CLIMBER_IDLE_MODE)
+                                .smartCurrentLimit((int) MotorConstants.NEO_CURRENT_LIMIT.in(Amps));
+
+                rightClimberConfig.inverted(ClimberConstants.RIGHT_CLIMBER_INVERTED)
+                                .idleMode(ClimberConstants.CLIMBER_IDLE_MODE)
+                                .smartCurrentLimit((int) MotorConstants.NEO_CURRENT_LIMIT.in(Amps));
+
+                leftClimberConfig.encoder.positionConversionFactor(ClimberConstants.CLIMBER_POSITION_CONVERSION_FACTOR)
+                                .velocityConversionFactor(ClimberConstants.CLIMBER_VELOCITY_CONVERSION_FACTOR);
+
+                leftClimberConfig.closedLoop.feedbackSensor(FeedbackSensor.kPrimaryEncoder)
+                                .pidf(ClimberConstants.P_VALUE, ClimberConstants.I_VALUE, ClimberConstants.D_VALUE,
+                                                ClimberConstants.CLIMBER_VELOCITY_FF);
+
+                leftClimberConfig.softLimit
+                                .forwardSoftLimit(ClimberConstants.UPPER_LIMIT.in(Inches))
+                                .forwardSoftLimitEnabled(true)
+                                .reverseSoftLimit(ClimberConstants.LOWER_LIMIT.in(Inches))
+                                .reverseSoftLimitEnabled(true);
+
+                rightClimberConfig.follow(leftClimber.getDeviceId(), true);
+
+                leftClimber.configure(leftClimberConfig, ResetMode.kResetSafeParameters,
+                                PersistMode.kPersistParameters);
+                rightClimber.configure(rightClimberConfig, ResetMode.kResetSafeParameters,
+                                PersistMode.kPersistParameters);
+
+                leftClimberEncoder.setPosition(ClimberConstants.LOWER_LIMIT.in(Inches));
+        }
+
+        public void setSpeed(LinearVelocity speed) {
+                climberClosedLoopController.setReference(speed.in(InchesPerSecond), ControlType.kVelocity);
+                if (speed.equals(InchesPerSecond.zero())) {
+                        climberServo.setAngle(90);
+                        servoGoalAngle = Degrees.of(90);
+                } else {
+                        climberServo.setAngle(270);
+                        servoGoalAngle = Degrees.of(270);
```

Comment:
:pick: we can reduce the delta between this and the "zero" angle to make the lock more responsive